test/JDBC/expected/sys-eomonth-vu-verify.out

```diff
@@ -0,0 +1,322 @@
+--Checking when separator is ‘-’ 
+--Checking valid date format for ‘YYYY-MM-DD’.
+SELECT EOMONTH ('1996-01-20')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for YYYY-MON-DD’.
+SELECT EOMONTH ('1996-JAN-20')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘MM-DD-YYYY’.
+SELECT EOMONTH ('01-20-1996')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘MM-DD-YY’.
+SELECT EOMONTH ('1-20-96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+
+--Checking when separator is ‘/’ 
+--Checking valid date format for ‘YYYY/MM/DD’.
+SELECT EOMONTH ('1996/01/20',30)
+GO
+~~START~~
+date
+1998-07-31
+~~END~~
+
+
+--Checking valid date format for YYYY/MON/DD’.
+SELECT EOMONTH ('1996/JAN/20',-30)
+GO
+~~START~~
+date
+1993-07-31
+~~END~~
+
+
+--Checking valid date format for ‘MM/DD/YYYY’.
+SELECT EOMONTH ('02/20/1996',100)
+GO
+~~START~~
+date
+2004-06-30
+~~END~~
+
+
+--Checking valid date format for ‘MM/DD/YY’.
+SELECT EOMONTH ('1/20/96',1200)
+GO
+~~START~~
+date
+2096-01-31
+~~END~~
+
+
+--Checking when separator is ‘.’
+--Checking valid date format for ‘YYYY.MM.DD’.
+SELECT EOMONTH ('1996.01.20',0)
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for YYYY.MON.DD’.
+SELECT EOMONTH ('1996.JAN.20',10)
+GO
+~~START~~
+date
+1996-11-30
+~~END~~
+
+
+--Checking valid date format for ‘MM.DD.YYYY’.
+SELECT EOMONTH ('02.20.1996')
+GO
+~~START~~
+date
+1996-02-29
+~~END~~
+
+
+--Checking valid date format for ‘MM.DD.YY’.
+SELECT EOMONTH ('1.20.96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘Month DD YYYY’.
+SELECT EOMONTH ('JAN 20 1996')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘Month DD YY’.
+SELECT EOMONTH ('JAN 20 96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘Month DD, YYYY’.
+SELECT EOMONTH ('JAN 20, 1996')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘Month DD, YY’.
+SELECT EOMONTH ('JAN 20, 96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘DD Month YYYY’.
+SELECT EOMONTH ('20 JAN 1996')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘DD Month YY’.
+SELECT EOMONTH ('20 JAN 96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘DD Month YYYY’.
+SELECT EOMONTH ('20 JAN, 1996')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘DD Month YY’.
+SELECT EOMONTH ('20 JAN, 96')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘YYYYMMDD’.
+SELECT EOMONTH ('19960120')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking valid date format for ‘YYMMDD’.
+SELECT EOMONTH ('960120')
+GO
+~~START~~
+date
+1996-01-31
+~~END~~
+
+
+--Checking for NULL it should return NULL
+SELECT EOMONTH (NULL)
```

Comment:
You may also explicitly test all combinations of NULL for your 2 arguments

Reply:
Yes, will add them in next revision.

Reply:
I have added the tests for all the combinations of NULL for your 2 arguments.

Reply:
I have added all possible combinations for NULL except 1 that is mentioned below. Because in SQL server It returns me the value for that months last date. In BBF the NULL is not getting casted to integer and so it's returning me NULL.
```
1> SELECT EOMONTH('1990-01-01',NULL)
2> go
eomonth         
------------
         NULL
```